### PR TITLE
Implement TextInputPlugin on Android

### PR DESCRIPTION
**Pull request by abarth**
This plugin will eventually replace keyboard.mojom.

### CODE REVIEW

**Comment by abarth** [comment]
@jason-simmons @chinmaygarde

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        JSONArray args = message.getJSONArray("args");
+        if (method.equals("TextInput.show")) {
+            showTextInput(view);
+        } else if (method.equals("TextInput.hide")) {
+            hideTextInput(view);
+        } else if (method.equals("TextInput.setClient")) {
+            setTextInputClient(view, args.getInt(0), args.getJSONObject(0));
+        } else if (method.equals("TextInput.setEditingState")) {
+            setTextInputEditingState(view, args.getJSONObject(0));
+        } else if (method.equals("TextInput.clearClient")) {
```
Comment (Hixie) [inline shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java]: it's really sad that we have to do string compares for method dispatch

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            clearTextInputClient();
+        } else {
+            // TODO(abarth): We should throw an exception here that gets
```
Comment (Hixie) [inline shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java]: why don't we?

**Comment by Hixie** [comment]
RSLGTM

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java (hunk 2 above)
Comment (abarth) [inline shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java]: I haven't built the exception handler logic yet.  :)

**Pull request merged**